modules/features/manatime_feature_timesheets/manatime_feature_timesheets.module

```diff
@@ -50,3 +84,273 @@ function manatime_feature_timesheets_cron() {
+    // Get only the tasks not recorded in the reports.
+    if (empty($timesheet->getTimeEntriesPerTask($task->nid))) {
+      $list[$task->nid] = $task->title;
+    }
+  }
+  return $list;
+}
+
+/**
+ * Ajax menu callback - Add row.
+ */
+function manatime_feature_timesheets_add_row_callback($node, $ajax) {
+  if ($ajax) {
+    ctools_include('ajax');
+    ctools_include('modal');
+
+    $form_state = array(
+      'ajax' => TRUE,
+      'title' => t('Add new row'),
+    );
+    $form_state['build_info']['args'] = array($node);
+
+    // Use ctools to generate ajax instructions for the browser to create
+    // a form in a modal popup.
+    $output = ctools_modal_form_wrapper('manatime_feature_timesheets_add_row_form', $form_state);
+
+    // If the form has been submitted, there may be additional instructions
+    // such as dismissing the modal popup.
+    if (!empty($form_state['ajax_commands'])) {
+      $output = $form_state['ajax_commands'];
+    }
+
+    // Return the ajax instructions to the browser via ajax_render().
+    print ajax_render($output);
+    drupal_exit();
+  }
+  else {
+    return drupal_get_form('manatime_feature_timesheets_add_row_form');
+  }
+}
+
+/**
+ * Ajax menu callback - Remove row.
+ */
+function manatime_feature_timesheets_remove_row_callback($node, $task_id, $ajax) {
+  if ($ajax) {
+    ctools_include('ajax');
+    ctools_include('modal');
+
+    $form_state = array(
+      'ajax' => TRUE,
+      'title' => t('Remove row'),
+    );
+
+    $form_state['build_info']['args'] = array($node, $task_id);
+
+    // Use ctools to generate ajax instructions for the browser to create
+    // a form in a modal popup.
+    $output = ctools_modal_form_wrapper('manatime_feature_timesheets_remove_row_form', $form_state);
+
+    // If the form has been submitted, there may be additional instructions
+    // such as dismissing the modal popup.
+    if (!empty($form_state['ajax_commands'])) {
+      $output = $form_state['ajax_commands'];
+    }
+
+    // Return the ajax instructions to the browser via ajax_render().
+    print ajax_render($output);
+    drupal_exit();
+  }
+  else {
+    return drupal_get_form('manatime_feature_timesheets_remove_row_form');
+  }
+}
+
+/**
+ * Update the section to the dropdown task.
+ */
+function refresh_tasks_callback($form, $form_state) {
+  return $form['task'];
+}
+
+/**
+ * Drupal form submit handler.
+ */
+function manatime_feature_timesheets_add_row_form_submit(&$form, &$form_state) {
+  // Generate the new link using the submitted text value.
+  $node = $form_state['build_info']['args'][0];
+  $timesheet = new Timesheet($node);
+  $timesheet->generateRecords();
+  $project = entity_metadata_wrapper('node', $form_state['complete form']['project']['#value']);
+  $task = entity_metadata_wrapper('node', $form_state['complete form']['task']['#value']);
+  // Add new row.
+  $remove_link_row = manatime_feature_timesheet_make_link_remove_row($node, $task->nid);
```

Comment:
This is not working because $task is a metadata wrapper so, you can't access nid. Use $task->getIdentifier();